request_test.go

```diff
@@ -318,6 +321,29 @@ func TestDoWithStringResponse(t *testing.T) {
 	})
 }
 
+func TestDoWithCustomClient(t *testing.T) {
+	customClient := GetClient()
+
+	jar, err := cookiejar.New(nil)
+	require.NoError(t, err)
+	customClient.Jar = jar
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Set-Cookie", "foo=bar")
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	u, err := url.Parse(ts.URL)
```

Comment:
Why are you not using ts.URL directly in the params? (see https://golang.org/pkg/net/http/httptest/#Server)